compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -326,6 +324,23 @@ BLangStatementExpression desugar(BLangQueryExpr queryExpr, SymbolEnv env,
         return streamStmtExpr;
     }
 
+    private BType getMapType(BType type) {
+        BType refType = Types.getReferredType(type);
+        if (refType.tag == TypeTags.UNION) {
+            for (BType memberType : ((BUnionType) type).getMemberTypes()) {
+                BType tempType = getMapType(memberType);
+                if (tempType != symTable.noType) {
+                    return tempType;
+                }
+            }
```

Comment:
Can we check and add few tests for the uncovered paths

Reply:
Seems we have to change the logic similar to https://github.com/ballerina-platform/ballerina-lang/pull/36743/commits/cfdb025ccf56d098e7b485b4a583d7e86e31527e. But I changed that logic to address the review suggestions. Shall I revert the changes? @pcnfernando

Reply:
This depends on when noType be returned https://github.com/ballerina-platform/ballerina-lang/pull/36743/files#r938427461

Reply:
But it is already covered with tests. I debug and check few examples.
I noticed that the `mapType` is placed earlier compared `errorType` inside the `memberType` list in the `unionType`. Therefore, seems we cannot cover this line in current implementation. But I think there is a possibility to fill up the `memberTypes` in `unionType` in different order. Therefore, we have to keep this logic. isn't it? @pcnfernando ?

Reply:
The type of the query expr resolved via `resolveQueryType()` method creates the union at this line right? https://github.com/ballerina-platform/ballerina-lang/blob/a2c659217a59162a7deb54bcf001a1c44bedbd34/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L5658
The `completionType` (which is `error` type here) will anyway come after the `actualType` when creating the union.? :thinking:

Reply:
Yeah. Updated in https://github.com/ballerina-platform/ballerina-lang/pull/36743/commits/5b3e2ea4988362bb08b7ec1054dbe1a13eb7ffc4.